test/performance/odp_ipsec.c

```diff
@@ -643,41 +655,58 @@ run_measure_one(ipsec_args_t *cargs,
 
 	while ((packets_sent < cargs->iteration_count) ||
 	       (packets_received < cargs->iteration_count)) {
-		if ((packets_sent < cargs->iteration_count) &&
+		int num_out = cargs->iteration_count - packets_sent;
+
+		num_out = num_out > burst_sz ? burst_sz : num_out;
+
+		if (num_out &&
 		    (packets_sent - packets_received <
 		     cargs->in_flight)) {
-			odp_packet_t out_pkt;
-			int num_out = 1;
+			odp_packet_t out_pkt[POOL_NUM_PKT];
+			int i;
 
-			pkt = make_packet(pkt_pool, payload_length);
-			if (ODP_PACKET_INVALID == pkt)
+			if (odp_unlikely(make_packet_multi(pkt_pool,
+							   payload_length,
+							   pkt,
+							   num_out)))
 				return -1;
 
-			if (cargs->debug_packets)
-				odp_packet_print_data(pkt, 0,
-						      odp_packet_len(pkt));
+			if (cargs->debug_packets) {
+				for (i = 0; i < num_out; i++)
+					odp_packet_print_data(pkt[i], 0,
+							      odp_packet_len(pkt[i]));
+			}
 
-			rc = odp_ipsec_out(&pkt, 1,
-					   &out_pkt, &num_out,
+			i = num_out;
+
+			rc = odp_ipsec_out(pkt, num_out,
+					   out_pkt, &num_out,
 					   &param);
 			if (rc <= 0) {
 				ODPH_ERR("failed odp_ipsec_out: rc = %d\n", rc);
-				odp_packet_free(pkt);
+				odp_packet_free_sp(pkt, i);
 				break;
 			}
-			if (odp_packet_has_error(out_pkt)) {
-				odp_ipsec_packet_result_t result;
 
-				odp_ipsec_result(&result, out_pkt);
-				ODPH_ERR("Received error packet: %d\n",
-					 result.status.error.all);
+			for (i = 0; i < num_out; i++) {
+				if (odp_packet_has_error(out_pkt[i])) {
```

Comment:
I think checking odp_packet_has_error() here and not checking the IPsec packet result here and in the async function is wrong. But maybe that needs to be fixed in a different patch.

Reply:
I made PR 1506 to fix this.